src/main/java/io/cryostat/agent/Registration.java

```diff
@@ -150,26 +133,32 @@ private void tryUpdate(Long id) {
             return;
         }
         log.info("publishing self as {}", selfNode.getTarget().getConnectUrl());
-        cryostat.update(pluginInfo, Set.of(selfNode))
-                .onSuccess(
-                        ar -> {
-                            if (id != null) {
-                                getVertx().cancelTimer(id);
-                            }
-                        })
-                .onFailure(
-                        t -> {
-                            log.error("Update failure", t);
-                            deregister()
-                                    .onComplete(
-                                            ar -> {
-                                                if (ar.failed()) {
-                                                    vertx.setTimer(
-                                                            registrationRetryMs, this::tryRegister);
-                                                    return;
-                                                }
-                                            });
-                        });
+        Future<Void> f =
+                cryostat.update(pluginInfo, Set.of(selfNode))
+                        .handleAsync(
+                                (n, t) -> {
+                                    if (n != null) {
```

Comment:
> and the executor t == null

Not sure if you meant `t` (Throwable) or the `executor` -

The executor should simply never be null. If it is this NPEs and fails, and that's that.

If both `n == null && t == null` then I guess I can just log something, since that should never happen.